### ISSUE TITLE
[kubernetes] Error restarting flowbroker when remote node exists

### ISSUE THREAD

**Issue by betefaber** [dojot/dojot#1348]
Error restarting flowbroker in kubernetes environment.

**kubernetes environment:**
```
<19:01:46 02/01/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment adminkelvin...
<19:01:46 02/01/2020> -- |kb8sRemoveNode -- DEBUG: Removing container adminkelvin from the set.
<19:01:46 02/01/2020> -- |kb8sRemoveNode -- DEBUG: Current container list is 
<19:01:46 02/01/2020> -- |kb8sRemoveNode -- DEBUG: Scaling down deployment flownode-adminkelvin...
<19:01:46 02/01/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment scale down request was sent to the server.
<19:01:46 02/01/2020> -- ERROR: Unhandled Rejection at: Error: deployments.extensions "flownode-adminkelvin" is forbidden: User "system:serviceaccount:dojot:flowbroker" cannot patch resource "deployments" in API group "extensions" in the namespace "dojot"
    at /opt/flowbroker/orchestrator/node_modules/kubernetes-client/lib/base.js:11:21
    at Request.request [as _callback] (/opt/flowbroker/orchestrator/node_modules/kubernetes-client/lib/request.js:107:14)
    at Request.self.callback (/opt/flowbroker/orchestrator/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/opt/flowbroker/orchestrator/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/opt/flowbroker/orchestrator/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9). Bailing out!!
```

![Screenshot at 2020-01-02 16-03-19](https://user-images.githubusercontent.com/37310063/71686801-f7826780-2d7a-11ea-9257-0a76840a6401.png)
----

Error does not occurs in docker-compose environment.

**docker-compose environment:**
```
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: Loading remote nodes for tenant admin ...
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: Loading remote nodes for tenant teste ...
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: Loading remote node {"_id":"5e0e46c9e017d600062ee025","id":"kelvin","image":"dojot/kelvin-example:3.0.0-alpha2","containerId":"2e27a7029745079ba9dae9c7b9418449f4ded18ed2d339395f5c1ee1f80d0811","metadata":{"id":"sample/kelvin","name":"kelvin","module":"kelvin","version":"0.0.0"}}.
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: Trying to remove docker container 2e27a7029745079ba9dae9c7b9418449f4ded18ed2d339395f5c1ee1f80d0811
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: ...Loaded remote nodes for tenant admin.
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: Succeeded to set manager to handle processing nodes for tenant admin
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: ...Loaded remote nodes for tenant teste.
flowbroker_1                  | <20:31:23 02/01/2020> -- |nodeMngr -- DEBUG: Succeeded to set manager to handle processing nodes for tenant teste
flowbroker_1                  | <20:31:23 02/01/2020> -- DEBUG: Connections established with RabbitMQ!
flowbroker_1                  | <20:31:23 02/01/2020> -- |amqp -- INFO: Producer ready ... 
flowbroker_1                  | <20:31:23 02/01/2020> -- |amqp -- INFO: consumer [object Object] ready ... 
flowbroker_1                  | <20:31:23 02/01/2020> -- |amqp -- INFO: Producer ready ... 
flowbroker_1                  | <20:31:25 02/01/2020> -- DEBUG: Unsubscribing from topics cd6f0177-a4d2-4b98-8a00-abbe7b53989f
flowbroker_1                  | <20:31:25 02/01/2020> -- DEBUG: Subscribing to topics cd6f0177-a4d2-4b98-8a00-abbe7b53989f
flowbroker_1                  | <20:31:25 02/01/2020> -- DEBUG: List of topics changed in the last few seconds. Delaying again
flowbroker_1                  | <20:31:26 02/01/2020> -- |nodeMngr -- DEBUG: Succeeded to remove container.
flowbroker_1                  | <20:31:27 02/01/2020> -- DEBUG: Unsubscribing from topics ed0facde-9c6a-488b-bccd-bcbf460e1741,f506c193-f294-48bf-befd-b2c5583f3d28,42f063a3-58ef-462d-abb5-26ec3d5449b0,b6031055-51b4-499f-8e02-41bd1932ad45,1486e2fa-15f9-4f4c-9592-2fbe6c2ba2b1,88c30024-5e2a-4537-9460-d3cec3bb3946
flowbroker_1                  | <20:31:27 02/01/2020> -- DEBUG: Subscribing to topics ed0facde-9c6a-488b-bccd-bcbf460e1741,f506c193-f294-48bf-befd-b2c5583f3d28,42f063a3-58ef-462d-abb5-26ec3d5449b0,b6031055-51b4-499f-8e02-41bd1932ad45,1486e2fa-15f9-4f4c-9592-2fbe6c2ba2b1,88c30024-5e2a-4537-9460-d3cec3bb3946
flowbroker_1                  | [harbor-master][modem] error parsing body as json
flowbroker_1                  | <20:31:28 02/01/2020> -- |dockerRemoteNode -- DEBUG: [nodes] image dojot/kelvin-example:3.0.0-alpha2 created
flowbroker_1                  | <20:31:29 02/01/2020> -- |dockerRemoteNode -- DEBUG: [nodes] container flowbroker.adminkelvin.d9d7686 was created
flowbroker_1                  | <20:31:30 02/01/2020> -- |dockerRemoteNode -- DEBUG: Target: 719f9090b5af
flowbroker_1                  | <20:31:32 02/01/2020> -- |dockerRemoteNode -- DEBUG: [nodes] container up: flowbroker.adminkelvin.d9d7686:719f9090b5af4bc3ad2dea05c6457d2991c45d751d79c68034fdc5bd2871f195
flowbroker_1                  | <20:31:32 02/01/2020> -- |dispatcher -- INFO: Dispatcher connected to 719f9090b5af on port 5555
flowbroker_1                  | <20:31:32 02/01/2020> -- |dispatcher -- DEBUG: Received reply: {"payload":{"payload":{"id":"sample/kelvin","name":"kelvin","module":"kelvin","version":"0.0.0"}},"requestId":"83d1ef5d-3d46-467e-8a7e-9c5aa516b7ab"}
flowbroker_1                  | <20:31:32 02/01/2020> -- |dispatcher -- DEBUG: Received reply: {"payload":{"payload":"<script type=\"text/x-red\" data-template-name=\"kelvin\">\n    <div class=\"form-row\">\n        <label for=\"node-input-name\"><i class=\"fa fa-tag\"></i> Name</label>\n        <input type=\"text\" id=\"node-input-name\">\n    </div>\n    <div class=\"form-row\">\n        <label for=\"node-input-in\"><i class=\"fa fa-edit\"></i> <span>input from</span></label>\n        <input type=\"text\" id=\"node-input-in\" placeholder=\"payload\" style=\"width:250px;\">\n        <input type=\"hidden\" id=\"node-input-inFieldType\">\n    </div>\n    <div class=\"form-row\">\n        <label for=\"node-input-out\"><i class=\"fa fa-edit\"></i> <span>output to</span></label>\n        <input type=\"text\" id=\"node-input-out\" placeholder=\"payload\" style=\"width:250px;\">\n        <input type=\"hidden\" id=\"node-input-outFieldType\">\n    </div>\n</script>\n\n<script type=\"text/x-red\" data-help-name=\"kelvin\">\n</script>\n\n<script type=\"text/javascript\">\n    RED.nodes.registerType('kelvin', {\n        category: 'converters',\n        color: \"rgb(231, 231, 174)\",\n        defaults: {\n            name: { value: \"\", required: false },\n            out:  { value: \"payload\", validate: RED.validators.typedInput(\"outFieldType\") },\n            outFieldType: { value: \"msg\" },\n            in: { value: \"payload\", validate: RED.validators.typedInput(\"inFieldType\") },\n            inFieldType: { value: \"msg\" },\n        },\n        inputs: 1,\n        outputs: 1,\n        icon: \"cog.png\",\n        label: function () {\n            return this.name || \"to kelvin\";\n        },\n        labelStyle: function () {\n            return this.name ? \"node_label_italic\" : \"\";\n        },\n        oneditprepare: function () {\n            if (!this.outFieldType) {\n                this.outFieldType = 'msg';\n            }\n\n            $(\"#node-input-out\").typedInput({\n                default: 'msg',\n                types: ['msg'],\n                typeField: $(\"#node-input-outFieldType\")\n            });\n\n            if (!this.inFieldType) {\n                this.inFieldType = 'msg';\n            }\n\n            $(\"#node-input-in\").typedInput({\n                default: 'msg',\n                types: ['msg'],\n                typeField: $(\"#node-input-inFieldType\")\n            });\n        },\n        oneditsave: function () { }\n    });\n</script>"},"requestId":"cbc7afe2-135c-471c-837e-e28dc22d1704"}
```

Affected version: **v0.4.1_rc2**

**Comment by bizetti** [comment]
Ticket analysis (Jan, 3rd 2020): It could be fixed in a patch.

**Comment by betefaber** [comment]
The same behavior is observed in the removal of the node:

- adding remote node
```
<12:13:24 27/01/2020> -- |kb8sRemoveNode -- DEBUG: Building deployment creation request...
<12:13:24 27/01/2020> -- |kb8sRemoveNode -- DEBUG: Adding container dojot/kelvin-example:3.0.0-alpha2 to the set...
<12:13:24 27/01/2020> -- |kb8sRemoveNode -- DEBUG: ... container adminkelvin was added to the set.
<12:13:24 27/01/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment creation request was built.
<12:13:24 27/01/2020> -- |dispatcher -- INFO: Dispatcher connected to flownode-adminkelvin on port 5555
<12:13:29 27/01/2020> -- |nodeMngr -- DEBUG: Succeeded to add remote node with id kelvin
```

- remove remote node
```
DELETE {{HOST}}:{{PORT}}/flows/v1/node/kelvin

An invalid response was received from the upstream server
```
```
<12:28:06 27/01/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment adminkelvin...
<12:28:06 27/01/2020> -- |kb8sRemoveNode -- DEBUG: Removing container adminkelvin from the set.
<12:28:06 27/01/2020> -- |kb8sRemoveNode -- DEBUG: Current container list is 
<12:28:06 27/01/2020> -- |kb8sRemoveNode -- DEBUG: Scaling down deployment flownode-adminkelvin...
<12:28:06 27/01/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment scale down request was sent to the server.
<12:28:06 27/01/2020> -- ERROR: Unhandled Rejection at: Error: deployments.extensions "flownode-adminkelvin" is forbidden: User "system:serviceaccount:dojot:flowbroker" cannot patch resource "deployments" in API group "extensions" in the namespace "dojot"
    at /opt/flowbroker/orchestrator/node_modules/kubernetes-client/lib/base.js:11:21
    at Request.request [as _callback] (/opt/flowbroker/orchestrator/node_modules/kubernetes-client/lib/request.js:107:14)
    at Request.self.callback (/opt/flowbroker/orchestrator/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/opt/flowbroker/orchestrator/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/opt/flowbroker/orchestrator/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9). Bailing out!!
```

**Comment by betefaber** [comment]
- initial status

![sem Kelvin](https://user-images.githubusercontent.com/37310063/73456869-6db1c400-4351-11ea-9df7-fd56a740792f.png)

![inicial](https://user-images.githubusercontent.com/37310063/73456878-730f0e80-4351-11ea-982c-ce9ee70eb8e2.png)

- **adding remote node**
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{"message":"ok"}
```
```
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Building deployment creation request...
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Adding container dojot/kelvin-example:3.0.0-alpha2 to the set...
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: ... container adminkelvin was added to the set.
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment creation request was built.
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Deployment is:
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: { apiVersion: 'extensions/v1beta1',
  kind: 'Deployment',
  metadata: 
   { labels: { app: 'flowbroker', name: 'flownode-adminkelvin' },
     name: 'flownode-adminkelvin' },
  spec: 
   { replicas: 1,
     template: 
      { metadata: { labels: { name: 'flownode-adminkelvin', app: 'flowbroker' } },
        spec: 
         { containers: 
            [ { name: 'adminkelvin',
                image: 'dojot/kelvin-example:3.0.0-alpha2',
                imagePullPolicy: 'Always',
                ports: [ { name: 'amqp', port: 5555, containerPort: 5555 } ] } ],
           restartPolicy: 'Always' } } } }
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Creating service for this deployment...
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Service to be created: { apiVersion: 'v1',
  kind: 'Service',
  metadata: 
   { name: 'flownode-adminkelvin',
     labels: { app: 'flowbroker', name: '' } },
  spec: 
   { selector: { name: 'flownode-adminkelvin' },
     ports: [ { protocol: 'TCP', port: 5555, targetPort: 5555 } ] } }
<13:56:30 30/01/2020> -- |kb8sRemoveNode -- DEBUG: ... service for deployment created:  { kind: 'Service',
  apiVersion: 'v1',
  metadata: 
   { name: 'flownode-adminkelvin',
     namespace: 'dojot',
     selfLink: '/api/v1/namespaces/dojot/services/flownode-adminkelvin',
     uid: '4b28a7aa-4bbd-49e3-a08f-4f139a3d86e3',
     resourceVersion: '12911584',
     creationTimestamp: '2020-01-30T13:56:30Z',
     labels: { app: 'flowbroker', name: '' } },
  spec: 
   { ports: [ { protocol: 'TCP', port: 5555, targetPort: 5555 } ],
     selector: { name: 'flownode-adminkelvin' },
     clusterIP: '10.111.193.7',
     type: 'ClusterIP',
     sessionAffinity: 'None' },
  status: { loadBalancer: {} } }
<13:56:30 30/01/2020> -- |dispatcher -- INFO: Dispatcher connected to flownode-adminkelvin on port 5555
<13:56:36 30/01/2020> -- |nodeMngr -- DEBUG: Succeeded to add remote node with id kelvin
```
![Screenshot at 2020-01-30 10-57-23](https://user-images.githubusercontent.com/37310063/73456903-828e5780-4351-11ea-8ec9-4b2ead220bbf.png)

![Screenshot at 2020-01-30 11-25-44](https://user-images.githubusercontent.com/37310063/73458112-8622de00-4353-11ea-94d5-5ef10f3a4081.png)

- **restart flowbroker**
```
cpqd@dojot-master:~$ kubectl -n dojot delete pod/flowbroker-7b88b69df7-c5kfp
pod "flowbroker-7b88b69df7-c5kfp" deleted
```
![restart](https://user-images.githubusercontent.com/37310063/73456957-9a65db80-4351-11ea-976d-17fa0f647340.png)

```
<14:03:54 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Adding container dojot/kelvin-example:3.0.0-alpha2 to the set...
<14:03:54 30/01/2020> -- |kb8sRemoveNode -- DEBUG: ... container adminkelvin was added to the set.
<14:03:54 30/01/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment creation request was built.
<14:03:54 30/01/2020> -- |kb8sRemoveNode -- DEBUG: Deployment is:
<14:03:54 30/01/2020> -- |kb8sRemoveNode -- DEBUG: { apiVersion: 'extensions/v1beta1',
  kind: 'Deployment',
  metadata: 
   { labels: { app: 'flowbroker', name: 'flownode-adminkelvin' },
     name: 'flownode-adminkelvin' },
  spec: 
   { replicas: 1,
     template: 
      { metadata: { labels: { name: 'flownode-adminkelvin', app: 'flowbroker' } },
        spec: 
         { containers: 
            [ { name: 'adminkelvin',
                image: 'dojot/kelvin-example:3.0.0-alpha2',
                imagePullPolicy: 'Always',
                ports: [ { name: 'amqp', port: 5555, containerPort: 5555 } ] } ],
           restartPolicy: 'Always' } } } }
<14:03:54 30/01/2020> -- |dispatcher -- INFO: Dispatcher connected to undefined on port 5555
<14:03:54 30/01/2020> -- DEBUG: Connections established with RabbitMQ!
<14:03:54 30/01/2020> -- |amqp -- INFO: Producer ready ... 
<14:03:54 30/01/2020> -- |amqp -- INFO: Producer ready ... 
<14:03:54 30/01/2020> -- |amqp -- INFO: consumer [object Object] ready ... 
<14:03:56 30/01/2020> -- DEBUG: Unsubscribing from topics 993917a0-c54f-41a7-8a66-6d33175cf6fa
<14:03:56 30/01/2020> -- DEBUG: Subscribing to topics 993917a0-c54f-41a7-8a66-6d33175cf6fa
<14:03:56 30/01/2020> -- DEBUG: List of topics changed in the last few seconds. Delaying again
<14:03:58 30/01/2020> -- DEBUG: Unsubscribing from topics c7a912ef-084d-4048-b102-6dcbb0dc12ce,a6b5b784-7015-4f4f-a3db-9b70ad92d430,e346a594-b5c5-4215-be81-67bffd4b7b22
<14:03:58 30/01/2020> -- DEBUG: Subscribing to topics c7a912ef-084d-4048-b102-6dcbb0dc12ce,a6b5b784-7015-4f4f-a3db-9b70ad92d430,e346a594-b5c5-4215-be81-67bffd4b7b22
```
Although the message states that the node has been added, it is not shown in the GUI

![Screenshot at 2020-01-30 11-06-21](https://user-images.githubusercontent.com/37310063/73456970-a356ad00-4351-11ea-8932-64e87c5f8eb4.png)

**Comment by betefaber** [comment]
The remote node was added to the database:
```
curl -X GET "${DOJOT_HOST}/flows/v1/node" -H "Authorization: Bearer ${JWT}" -H "Content-Type:application/json"
{
    "nodes": [
        {
            "id": "kelvin",
            "image": "dojot/kelvin-example:3.0.0-alpha2"
        }
    ]
}
```
but does not appear in the GUI:

![image](https://user-images.githubusercontent.com/37310063/73543427-b59a1f00-4415-11ea-88f4-cb00041f4b74.png)

**Comment by bizetti** [comment]
Ticket analysis (Jan, 31st): Try to add new node and verify if it is sucessful.

**Comment by betefaber** [comment]
- **add the remote node again**: the node is not added
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
The upstream server is timing out
```
- **failed to add the remote node with another id**: the container is created but the node is not added
```
curl -X POST ${DOJOT_HOST}/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin2"}'
{"message":"Failed to add node: Failed to initiate docker container."}
```
```
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: Building deployment creation request...
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: Adding container dojot/kelvin-example:3.0.0-alpha2 to the set...
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: ... container adminkelvin2 was added to the set.
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment creation request was built.
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: Deployment is:
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: { apiVersion: 'extensions/v1beta1',
  kind: 'Deployment',
  metadata: 
   { labels: { app: 'flowbroker', name: 'flownode-adminkelvin2' },
     name: 'flownode-adminkelvin2' },
  spec: 
   { replicas: 1,
     template: 
      { metadata: 
         { labels: { name: 'flownode-adminkelvin2', app: 'flowbroker' } },
        spec: 
         { containers: 
            [ { name: 'adminkelvin2',
                image: 'dojot/kelvin-example:3.0.0-alpha2',
                imagePullPolicy: 'Always',
                ports: [ { name: 'amqp', port: 5555, containerPort: 5555 } ] } ],
           restartPolicy: 'Always' } } } }
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: Creating service for this deployment...
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: Service to be created: { apiVersion: 'v1',
  kind: 'Service',
  metadata: 
   { name: 'flownode-adminkelvin2',
     labels: { app: 'flowbroker', name: '' } },
  spec: 
   { selector: { name: 'flownode-adminkelvin2' },
     ports: [ { protocol: 'TCP', port: 5555, targetPort: 5555 } ] } }
<17:14:08 05/02/2020> -- |kb8sRemoveNode -- DEBUG: ... service for deployment created:  { kind: 'Service',
  apiVersion: 'v1',
  metadata: 
   { name: 'flownode-adminkelvin2',
     namespace: 'dojot',
     selfLink: '/api/v1/namespaces/dojot/services/flownode-adminkelvin2',
     uid: 'f403fc51-6827-431b-9e47-36ec34c76b8d',
     resourceVersion: '13139803',
     creationTimestamp: '2020-02-05T17:14:08Z',
     labels: { app: 'flowbroker', name: '' } },
  spec: 
   { ports: [ { protocol: 'TCP', port: 5555, targetPort: 5555 } ],
     selector: { name: 'flownode-adminkelvin2' },
     clusterIP: '10.98.63.171',
     type: 'ClusterIP',
     sessionAffinity: 'None' },
  status: { loadBalancer: {} } }
<17:14:08 05/02/2020> -- |dispatcher -- INFO: Dispatcher connected to flownode-adminkelvin2 on port 5555
<17:14:13 05/02/2020> -- |dispatcher -- DEBUG: Received reply: {"payload":{"payload":{"id":"sample/kelvin","name":"kelvin","module":"kelvin","version":"0.0.0"}},"requestId":"14381a2c-f0c3-40a2-8717-44fbfcb4d197"}
<17:14:13 05/02/2020> -- |dispatcher -- DEBUG: Received reply: {"payload":{"payload":"<script type=\"text/x-red\" data-template-name=\"kelvin\">\n    <div class=\"form-row\">\n        <label for=\"node-input-name\"><i class=\"fa fa-tag\"></i> Name</label>\n        <input type=\"text\" id=\"node-input-name\">\n    </div>\n    <div class=\"form-row\">\n        <label for=\"node-input-in\"><i class=\"fa fa-edit\"></i> <span>input from</span></label>\n        <input type=\"text\" id=\"node-input-in\" placeholder=\"payload\" style=\"width:250px;\">\n        <input type=\"hidden\" id=\"node-input-inFieldType\">\n    </div>\n    <div class=\"form-row\">\n        <label for=\"node-input-out\"><i class=\"fa fa-edit\"></i> <span>output to</span></label>\n        <input type=\"text\" id=\"node-input-out\" placeholder=\"payload\" style=\"width:250px;\">\n        <input type=\"hidden\" id=\"node-input-outFieldType\">\n    </div>\n</script>\n\n<script type=\"text/x-red\" data-help-name=\"kelvin\">\n</script>\n\n<script type=\"text/javascript\">\n    RED.nodes.registerType('kelvin', {\n        category: 'converters',\n        color: \"rgb(231, 231, 174)\",\n        defaults: {\n            name: { value: \"\", required: false },\n            out:  { value: \"payload\", validate: RED.validators.typedInput(\"outFieldType\") },\n            outFieldType: { value: \"msg\" },\n            in: { value: \"payload\", validate: RED.validators.typedInput(\"inFieldType\") },\n            inFieldType: { value: \"msg\" },\n        },\n        inputs: 1,\n        outputs: 1,\n        icon: \"cog.png\",\n        label: function () {\n            return this.name || \"to kelvin\";\n        },\n        labelStyle: function () {\n            return this.name ? \"node_label_italic\" : \"\";\n        },\n        oneditprepare: function () {\n            if (!this.outFieldType) {\n                this.outFieldType = 'msg';\n            }\n\n            $(\"#node-input-out\").typedInput({\n                default: 'msg',\n                types: ['msg'],\n                typeField: $(\"#node-input-outFieldType\")\n            });\n\n            if (!this.inFieldType) {\n                this.inFieldType = 'msg';\n            }\n\n            $(\"#node-input-in\").typedInput({\n                default: 'msg',\n                types: ['msg'],\n                typeField: $(\"#node-input-inFieldType\")\n            });\n        },\n        oneditsave: function () { }\n    });\n</script>"},"requestId":"d673bb6a-1df0-47de-ad45-76d9d1754ae4"}
<17:14:13 05/02/2020> -- |nodeMngr -- ERROR: Failed to remove remote node
                admin/kelvin2 (Error: Tenant undefined has not been initialized.). keep going ..
<17:14:13 05/02/2020> -- |nodeMngr -- ERROR: Failed to initiate docker container for
            remote node admin/kelvin2 (ReferenceError: name is not defined).
<17:14:13 05/02/2020> -- |nodeMngr -- ERROR: Failed to remove remote node
              admin/kelvin2 (Error: Tenant undefined has not been initialized.). keep going ..
```

![Screenshot at 2020-02-05 14-18-00](https://user-images.githubusercontent.com/37310063/73868242-e4096700-4826-11ea-9ce0-19b42b7c0124.png)

![Screenshot at 2020-02-05 14-18-27](https://user-images.githubusercontent.com/37310063/73868264-e966b180-4826-11ea-939a-63d292bc0d01.png)

**Comment by bizetti** [comment]
Ticket analysis (Feb, 14th): It's possible that not invoking dinit in kubernets environment and invoking init, the problem will be solved (method _loadRemoteNodes).

**Comment by betefaber** [comment]
Fixed in v0.4.3-rc.1

![Screenshot_78](https://user-images.githubusercontent.com/37310063/90186298-52a71e80-dd8e-11ea-9b3e-78c7d556d9fc.png)

**Comment by betefaber** [comment]
Fixed in v0.4.3-rc.1.

- remote node exists

![Screenshot_79](https://user-images.githubusercontent.com/37310063/90281752-82fcc480-de43-11ea-9221-c3f78ad25c58.png)

- restarting flowbroker

![restartingflowbroker.png](https://images.zenhubusercontent.com/5cd0940469cf6b4ee443403a/58826b8d-9572-46be-96f3-475769ad0b5e)

- log

![logflowbroker.png](https://images.zenhubusercontent.com/5cd0940469cf6b4ee443403a/97aa1ab2-1c97-46c0-aadb-dc1e34ca096f)

**Comment by betefaber** [comment]
and the remote node is successfully removed
```
curl -X POST 10.50.11.181:30001/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json" -d '{"image": "dojot/kelvin-example:3.0.0-alpha2","id":"kelvin"}'
{"message":"ok"}
```
```
 curl -X GET 10.50.11.181:30001/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"nodes":[{"id":"kelvin","image":"dojot/kelvin-example:3.0.0-alpha2"}]}
```
![kelvinadicionado](https://user-images.githubusercontent.com/37310063/90287039-6e252e80-de4d-11ea-8c09-c3e6d10be352.png)

```
curl -X DELETE 10.50.11.181:30001/flows/v1/node/kelvin -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"message":"ok"}
```

```
<17:52:59 14/08/2020> -- |nodeMngr -- DEBUG: Trying to remove docker container undefined
<17:52:59 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Retrieving current deployment...
<17:52:59 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Sending request to server...
<17:52:59 14/08/2020> -- |kb8sRemoveNode -- DEBUG: ... request was sent to the server.
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment adminkelvin...
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Removing container adminkelvin from the set.
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Current container list is
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Scaling down deployment flownode-adminkelvin...
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment scale down request was sent to the server.
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment flownode-adminkelvin was scaled down.
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Removing deployment flownode-adminkelvin...
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment removal request was sent to the server.
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: ... deployment flownode-adminkelvin was removed.
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: Removing service for this deployment...
<17:53:00 14/08/2020> -- |kb8sRemoveNode -- DEBUG: ... Service for deployment removed:  { kind: 'Status',
  apiVersion: 'v1',
  metadata: {},
  status: 'Success',
  details:
   { name: 'flownode-adminkelvin',
     kind: 'services',
     uid: '17631773-53f1-4fb8-a6b3-8ffa74282987' } }
<17:53:00 14/08/2020> -- |nodeMngr -- DEBUG: Succeeded to remove container.
<17:53:00 14/08/2020> -- |nodeMngr -- DEBUG: Succeeded to delete remote node with id kelvin for tenant admin.
```
```
curl -X GET 10.50.11.181:30001/flows/v1/node -H "Authorization: Bearer ${JWT}" -H "content-type: application/json"
{"nodes":[]}
```
![removedkelvinteste](https://user-images.githubusercontent.com/37310063/90287054-7a10f080-de4d-11ea-94f3-bf3778d48eb0.png)
